src/components/Select/components/Control/Control.tsx

```diff
@@ -36,18 +58,54 @@ export const Control = (props: ControlProps) => {
         controlInlineStyles.width = width;
     }
 
+    const getControlNode = React.useCallback(() => {
+        return (ref as React.RefObject<HTMLButtonElement>)?.current;
+    }, [ref]);
+
+    const handleClick = () => {
+        dispatch({type: 'SET_ACTIVE', payload: {active: !active}});
+    };
+
+    const handleKeyDown = (e: React.KeyboardEvent<HTMLButtonElement>) => {
+        listboxRef?.current?.onKeyDown(e);
+    };
+
+    const handleResize = React.useCallback(() => {
+        const controlRect = getControlNode()?.getBoundingClientRect();
+        dispatch({type: 'SET_CONTROL_RECT', payload: {controlRect}});
+    }, [dispatch, getControlNode]);
+
+    React.useLayoutEffect(() => {
+        const resizeObserver = new ResizeObserver(handleResize);
+        const element = getControlNode();
+
+        if (element) {
+            resizeObserver.observe(element);
+        }
+
+        return () => {
+            if (element) {
+                resizeObserver.disconnect();
+            }
+        };
+    }, [getControlNode, handleResize]);
+
     return (
         <button
-            className={b(controlMods)}
+            ref={ref}
+            className={b(controlMods, className)}
             style={controlInlineStyles}
-            role="combobox"
```

Comment:
Why deleted the prop?

Reply:
@ValeraS said that we didn`t need that props :) but I couldn`t find it in documentation